addons/website_project/controllers/main.py

```diff
@@ -12,8 +12,15 @@ def insert_record(self, request, model, values, custom, meta=None):
         if model.model == 'project.task':
             visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
             visitor_partner = visitor_sudo.partner_id
-            if visitor_partner:
-                values['partner_id'] = visitor_partner.id
+            if not visitor_partner:
+                visitor_partner = visitor_partner.sudo().create({
+                    'email': request.params.get('email_from', False),
+                    'name': request.params.get('partner_name', False),
+                    'phone': request.params.get('partner_phone', False),
+                    'company_name': request.params.get('partner_company_name', False),
+                    'lang': request.lang.code,
+                })
```

Comment:
```suggestion
                def _prepare_visitor_partner_vals(self):
                  return {
                      'email': request.params.get('email_from', False),
                      'name': request.params.get('partner_name', False),
                      'phone': request.params.get('partner_phone', False),
                      'company_name': request.params.get('partner_company_name', False),
                      'lang': request.lang.code,
                  }
                  visitor_partner_vals = .._prepare_visitor_partner_vals()
                   values['partner_id'] = visitor_partner.sudo().create(visitor_partner_vals).id
```

code is not tested and create separate vals method so anyone can easily customization.